### PR TITLE
Add support for [GenerateImmutable] on sealed classes. Fixes #54.

### LINKED ISSUE #54: [GenerateImmutable] on sealed classes generates code with errors.
For example, the following will produce generated code with errors and warnings. The errors are due to adding virtual or protected members to a sealed class.
```
[GenerateImmutable( DefineInterface = true, DefineWithMethodsPerProperty = true, GenerateBuilder = true )]
sealed partial class Sealed
{
    readonly string name;
}
```

### PR DESCRIPTION
**Pull request by tg73**

### CODE REVIEW

File: src/ImmutableObjectGraph.CodeGeneration.Roslyn/CodeGen.cs [hunk 1]
```diff
@@ -355,13 +357,16 @@ private MemberDeclarationSyntax CreateCtor()
...
                 this.applyTo.Identifier)
-                .WithModifiers(SyntaxFactory.TokenList(SyntaxFactory.Token(SyntaxKind.ProtectedKeyword)))
                 .WithParameterList(
                     CreateParameterList(this.applyToMetaType.AllFields, ParameterStyle.Required)
                     .PrependParameter(RequiredIdentityParameter)
                     .AddParameters(SyntaxFactory.Parameter(SkipValidationParameterName.Identifier).WithType(SyntaxFactory.PredefinedType(SyntaxFactory.Token(SyntaxKind.BoolKeyword)))))
                 .WithBody(body);
 
+            if (!this.isSealed)
+                ctor = ctor
```
Comment (AArnott) [inline src/ImmutableObjectGraph.CodeGeneration.Roslyn/CodeGen.cs]: Please use C# blocks around all block statements.

**Comment by AArnott** [comment]
looks good after style corrections.

File: src/ImmutableObjectGraph.CodeGeneration.Tests/TestSources/Sealed.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+﻿namespace ImmutableObjectGraph.CodeGeneration.Tests.TestSources
+{
+    [GenerateImmutable( DefineInterface = true, DefineWithMethodsPerProperty = true, GenerateBuilder = true )]
```
Comment (AArnott) [inline src/ImmutableObjectGraph.CodeGeneration.Tests/TestSources/Sealed.cs]: This is the only file with non-VS default whitespace left after your first set of whitespace fixes.

**Comment by tg73** [comment]
Hi Andrew - is this pull request ok now? I've got another unrelated change in the works which has some code which needs to be sealed-conditional.

**Comment by AArnott** [comment]
Oh, sorry for the delay. yes, it looks great. Thanks.

**Pull request merged**